org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -576,6 +585,11 @@ private void updateSession() {
 
             session.setId(info.getSessionID());
             session.refreshData();
+        
+            NextProtoResult ret = SSL.GetNextProto(ssl_fd);
+            if (ret != null) {
+                session.setNextProtocol(ret.getProtocol());
```

Comment:
Needs to only be done when `SSL_NEXT_PROTO_NEGOTIATED` or `SSL_NEXT_PROTO_SELECTED`; other values should be `null`.